pkg/cluster/generateconfig.go

```diff
@@ -97,6 +98,12 @@ func (m *manager) generateInstallConfig(ctx context.Context) (*installconfig.Ins
 		masterZones = []string{""}
 	}
 
+	networkingNetworkType := string(m.doc.OpenShiftCluster.Properties.NetworkProfile.SDNProvider)
+	if networkingNetworkType == "" {
+		// No SDNProvider specified, default to this
+		networkingNetworkType = string(apiv20210131preview.SDNProviderOVNKubernetes)
```

Comment:
In `github.com/Azure/ARO-RP/pkg/cluster` we should not import anything from API versions: these values from external representation. Here we should work with internal representation only.

I imagine that we need something like:

```go
SDNProvider := "OVNKubernetes" // or "OpenShiftSDN", if we still want to have it be defaul
if m.doc.OpenShiftCluster.Properties.NetworkProfile.SDNProvider != "" {
    SDNProvider = m.doc.OpenShiftCluster.Properties.NetworkProfile.SDNProvider
}

// And then use SDNProvider to instantiate an `InstallConfig`.
```

Reply:
Alternatively, we need a generic way to set defaults. But given that at the moment we have only one use case where we need to set defaults, I'm not sure that it is worth doing something more complicated at this point.

Reply:
Ack, I've taken your example.